Content.Server/GameTicking/Rules/Components/NinjaRuleComponent.cs

```diff
@@ -0,0 +1,59 @@
+using Content.Server.Ninja.Systems;
+using Content.Server.Objectives;
+using Robust.Shared.Audio;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.List;
+
+namespace Content.Server.GameTicking.Rules.Components;
+
+[RegisterComponent, Access(typeof(NinjaSystem))]
+public sealed class NinjaRuleComponent : Component
+{
+    /// <summary>
+    /// List of objective prototype ids to add
+    /// </summary>
+    [DataField("objectives", required: true, customTypeSerializer: typeof(PrototypeIdListSerializer<ObjectivePrototype>))]
+    public readonly List<string> Objectives = new();
+
+    // TODO: move to job and use job???
+    /// <summary>
+    /// List of implants to inject on spawn
+    /// </summary>
+    [DataField("implants", required: true, customTypeSerializer: typeof(PrototypeIdListSerializer<EntityPrototype>))]
+    public readonly List<string> Implants = new();
```

Comment:
Move this and the AddImplant code and stuff to its own component (or a field in ImplantedComponent idk)

Reply:
how would i make the implanting work? (assuming field on ImplantedComponent)
```
var implanted = EnsureComp<ImplantedComponent>(ninja);
// wait its already ComponentInit get trolled
implanted.StartingImplants.Add("MicrobombImplant");
```

Reply:
Uhh is there a problem with moving this field there, subscribing to ComponentStartup and calling `AddImplant(implant)` or whatever

Reply:
yes but isnt componentstartup raised before the field can be changed

Reply:
Uhh Startup is raised after every other component has been initialized (contrary to Init which is non-deterministic in the order) if thats what you meant, you can use OnMapInit instead if you get some issues tho

Reply:
@deltanedas noob not doing review